src/setup/config.rs

```diff
@@ -0,0 +1,59 @@
+//! Utilities for node configuration.
+
+use std::{ffi::OsString, fs, path::PathBuf};
+
+use anyhow::Result;
+use serde::Deserialize;
+
+use crate::setup::constants::SETUP_CONFIG;
+
+/// Startup configuration for the node.
+#[derive(Debug, Clone, Default)]
+pub struct NodeConfig {
+    /// Setting this option to true will enable node logging to stdout.
+    pub log_to_stdout: bool,
+}
+
+/// Convenience struct for reading Ziggurat's configuration file.
+#[derive(Deserialize)]
+struct ConfigFile {
+    /// The absolute path of where to run the start command.
+    path: PathBuf,
+    /// The command to start the node.
+    start_command: String,
+}
+
+/// The node metadata read from Ziggurat's configuration file.
+#[derive(Debug, Clone)]
+pub struct NodeMetaData {
+    /// The absolute path of where to run the start command.
+    pub path: PathBuf,
+    /// The command to start the node.
+    pub start_command: OsString,
+    /// The arguments to the start command of the node.
+    pub start_args: Vec<OsString>,
+}
+
+impl NodeMetaData {
+    pub fn new(setup_path: PathBuf) -> Result<NodeMetaData> {
```

Comment:
```suggestion
    pub fn new(setup_path: &Path) -> Result<NodeMetaData> {
```
it will become `PathBuf` after the `join`